src/core/GAT.jl

```diff
@@ -69,109 +69,147 @@ end
   end
 end
 
-""" Signature for a generalized algebraic theory (GAT).
+""" Data structure for a generalized algebraic theory (GAT).
 """
-@auto_hash_equals struct Signature
+@auto_hash_equals struct Theory
   types::Vector{TypeConstructor}
   terms::Vector{TermConstructor}
   axioms::Vector{AxiomConstructor}
   aliases::Dict{Symbol, Symbol}
 end
 
-""" Typeclass = GAT signature + Julia-specific content.
+""" Typeclass = GAT + Julia-specific content.
 """
 struct Typeclass
   name::Symbol
   type_params::Vector{Symbol}
-  signature::Signature
+  theory::Theory
   functions::Vector{JuliaFunction}
 end
 
-struct SignatureBinding
+struct TheoryBinding
   name::Symbol
   params::Vector{Symbol}
 end
-struct SignatureHead
-  main::SignatureBinding
-  base::Vector{SignatureBinding}
-  SignatureHead(main, base=[]) = new(main, base)
+struct TheoryHead
+  main::TheoryBinding
+  base::Vector{TheoryBinding}
+  TheoryHead(main, base=[]) = new(main, base)
 end
 
-# Signatures
+# Theories
 ############
 
-""" Define a signature of a generalized algebraic theory (GAT).
+""" Define a generalized algebraic theory (GAT).
 
-Three kinds of things can go in the signature body:
+Five kinds of things can go in the theory body:
 
-1. Type constructors, indicated by the special type `TYPE`, e.g.
+1. Type constructors, indicated by the special type `TYPE`, e.g.,
    `Hom(X::Ob,Y::Ob)::TYPE`
 2. Term constructors, e.g.,
    `id(X::Ob)::Hom(X,X)`
 3. Julia functions operating on the term constructors to provide additional
    functionality
+4. Function aliases, e.g.,
+   `@op Hom :→`
+5. Equivalency axioms, e.g.,
```

Comment:
Let's call these "equality axioms" rather than "equivalency axioms". That would be the more standard term.

Reply:
Fixed in commit 1a3a27c